packages/plugin-vue/src/index.ts

```diff
@@ -330,6 +330,9 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
       // select corresponding block for sub-part virtual modules
       if (query.vue) {
         if (query.src) {
+          if (options.value.devServer?.watcher) {
+            options.value.devServer.watcher.add(filename)
+          }
```

Comment:
I think it should use this.addWatchFile instead, so it also works with Rollup build watch mode.